pkg/api/networkservice/mechanisms/wireguard/helpers.go

```diff
@@ -23,80 +23,101 @@ import (
 	"github.com/networkservicemesh/api/pkg/api/networkservice"
 )
 
-// Mechanism - a wireguard mechanism utility wrapper
-type Mechanism interface {
-	// SrcIP -  src ip
-	SrcIP() net.IP
-	// DstIP - dst ip
-	DstIP() net.IP
-	// SrcPublicKey - source public key
-	SrcPublicKey() string
-	// DstPublicKey - destination public key
-	DstPublicKey() string
-	// SrcPort - Source interface listening port
-	SrcPort() int
-	// SrcPort - Destination interface listening port
-	DstPort() int
-}
-
-type mechanism struct {
+// Mechanism is a wireguard mechanism helper
+type Mechanism struct {
 	*networkservice.Mechanism
 }
 
 // ToMechanism - convert unified mechanism to useful wrapper
-func ToMechanism(m *networkservice.Mechanism) Mechanism {
+func ToMechanism(m *networkservice.Mechanism) *Mechanism {
 	if m.Type == MECHANISM {
 		if m.GetParameters() == nil {
 			m.Parameters = map[string]string{}
 		}
-		return &mechanism{
+		return &Mechanism{
 			m,
 		}
 	}
 	return nil
 }
 
-func (m *mechanism) SrcIP() net.IP {
+// SrcIP returns source ip
+func (m *Mechanism) SrcIP() net.IP {
 	return net.ParseIP(m.GetParameters()[SrcIP])
 }
 
-func (m *mechanism) DstIP() net.IP {
+// SetSrcIP sets source ip
+func (m *Mechanism) SetSrcIP(ip net.IP) {
```

Comment:
It is often *very* convenient to have Set* methods return the *Mechanism... it allows you to SetParameters in fluent style inline.

Reply:
I like this idea :)